sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobTransferMetadataOptions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.DataMovement
+{
+    /// <summary>
+    ///  Represents the options for transferring metadata from the source to the destination storage resource.
+    /// </summary>
+    public class BlobTransferMetadataOptions
```

Comment:
Can we not have both? A sort of "preserve metadata but add/overwrite these values" situation? Not sure we need to have exclusively one or the other.

Reply:
You're right, I didn't think about that scenario. That someone wants to keep the existing metadata, and they want to add more metadata on.